app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardDatastoresStep/MappingWizardDatastoresStep.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm } from 'redux-form';
+import { noop, bindMethods } from 'patternfly-react';
+import SourceClusterSelect from './components/SourceClusterSelect/SourceClusterSelect';
+import DatastoresStepForm from './components/DatastoresStepForm';
+
+class MappingWizardDatastoresStep extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedCluster: undefined,
+      selectedClusterMapping: null
+    };
+
+    bindMethods(this, ['selectSourceCluster', 'resetState']);
+  }
+
+  componentDidMount() {}
+
+  selectSourceCluster(sourceClusterId) {
+    // when dropdown selection occurs for source cluster, we go retrieve the datastores for that
+    // cluster
+    const {
+      fetchDatastoresUrl,
+      fetchSourceDatastoresAction,
+      fetchTargetDatastoresAction,
+      clusterMappings
+    } = this.props;
+
+    const selectedClusterMapping = clusterMappings.find(clusterMapping =>
+      clusterMapping.nodes.some(
+        sourceCluster => sourceCluster.id === sourceClusterId
+      )
+    );
+
+    const { nodes: sourceClusters, ...targetCluster } = selectedClusterMapping;
+
+    this.setState(() => ({
```

Comment:
This is probably fine to leave as react component state, but I think we could move `selectedCluster` and `selectedClusterMapping` into redux state, and the remaining business logic in this component could be moved to the reducer.

However, can we do that if we are using redux-form to manage the state?  We may have to have some additional state values in the store outside the `form` property, if we want to move logic into redux. @priley86, @AparnaKarve, I'd be curious about your thoughts on that.

Again-- this is speaking to the optimal case, maybe we wait until post-POC to clean this up if it's working already, if we need to clean it up at all.